src/marqo/tensor_search/tensor_search.py

```diff
@@ -1929,45 +1934,104 @@ def _vector_text_search(
     else:
         contextualised_filter = ''
 
-    for vector_field in vector_properties_to_search:
-        search_query = {
-            "size": result_count,
-            "from": offset,
-            "query": {
-                "nested": {
-                    "path": TensorField.chunks,
-                    "inner_hits": {
-                        "_source": {
-                            "include": ["__chunks.__field_content", "__chunks.__field_name"]
-                        }
-                    },
-                    "query": {
-                        "knn": {
-                            f"{TensorField.chunks}.{vector_field}": {
-                                "vector": vectorised_text,
-                                "k": result_count + offset
+    if score_modifiers is not None:
+        validated_score_modifiers = validation.validate_score_modifiers_object(score_modifiers)
+        script_score = convert_validated_score_modifiers_to_script_score(validated_score_modifiers)
+        for vector_field in vector_properties_to_search:
```

Comment:
can we move this to separate functions, to prevent these functions getting too massive?

For example: `create_score_modified_tensor_search_body()`

Reply:
done